src/test/ReactTestUtils.js

```diff
@@ -213,14 +223,20 @@ var ReactTestUtils = {
     var internalInstance = ReactInstanceMap.get(inst);
     if (internalInstance && typeof internalInstance.tag === 'number') {
       var fiber = internalInstance;
-      var root = fiber;
-      while (root.return) {
-        root = root.return;
+      if (!fiber.child && fiber.alternate) {
+        // When we don't have children, we try the alternate first to see if it
+        // has any current children first.
+        fiber = fiber.alternate;
+      }
+      var results = findAllInRenderedFiberTreeInternal(fiber, test);
+      if (results === null) {
+        // Null is a sentinel that indicates that this was the wrong fiber.
+        results = findAllInRenderedFiberTreeInternal(fiber.alternate, test);
+        if (results === null) {
+          throw new Error('This should never happen.');
```

Comment:
Mm, I read it as an analog of this check:

https://github.com/facebook/react/blob/9da0bf53f78d06c65274fc039a6cfc3bd9d8b424/src/renderers/shared/fiber/ReactFiberTreeReflection.js#L100